### PR TITLE
fix(auth): Unauthenticated identity throws AuthError without user …

### LINKED ISSUE #9771: Amplify prints out AuthError when NOT having Auth component
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

REST API

### Amplify Categories

auth, api

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

Amplify JS version: "aws-amplify": "^4.3.17",

When setting up a Amplify React app WITHOUT Auth component, calling the API.get will print out the below error in browser console
```
[ERROR] 00:14.198 AuthError - 
            Error: Amplify has not been configured correctly. 
            The configuration object is missing required auth properties.
            This error is typically caused by one of the following scenarios:

            1. Did you run `amplify push` after adding auth via `amplify add auth`?
                See https://aws-amplify.github.io/docs/js/authentication#amplify-project-setup for more information

            2. This could also be caused by multiple conflicting versions of amplify packages, see (https://docs.amplify.aws/lib/troubleshooting/upgrading/q/platform/js) for help upgrading Amplify packages.
```

### Expected behavior

It should not print error in the browser console

### Reproduction steps

- set up a basic React app
- set up a public accessible API GW
- Use the following code snippet to call REST API
```jsx
import Amplify, { API } from 'aws-amplify';
Amplify.configure({
  API: {
    endpoints: [
      {
        name: "rest",
        endpoint: "<API endpoint>" 
      }
    ]
  }
});

function App() {

  API.get('rest', '/items')

  return (
    <div className="App">
     
    </div>
  );
}

export default App;
```

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

_No response_

### Manual configuration

_No response_

### Additional configuration

_No response_

### Mobile Device

_No response_

### Mobile Operating System

_No response_

### Mobile Browser

_No response_

### Mobile Browser Version

_No response_

### Additional information and screenshots

_No response_

### LINKED ISSUE #10039: Unauthenticated identity throws AuthError without user pool
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication, GraphQL API

### Amplify Categories

auth

### Environment information

<details>

```
# Put output below this line
npx: installed 1 in 1.189s

  System:
    OS: macOS 12.1
    CPU: (10) x64 Apple M1 Max
    Memory: 1.49 GB / 64.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.6 - ~/.nvm/versions/node/v14.17.6/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 6.14.15 - ~/.nvm/versions/node/v14.17.6/bin/npm
    Watchman: 2022.02.28.00 - /usr/local/bin/watchman
  Browsers:
    Brave Browser: 99.1.36.119
    Chrome: 103.0.5060.53
    Firefox: 100.0.1
    Safari: 15.2
  npmPackages:
    @ampproject/toolbox-optimizer:  undefined ()
    @babel/core:  undefined ()
    @babel/runtime:  7.15.4 
    @edge-runtime/primitives:  1.1.0-beta.10 
    @hapi/accept:  undefined ()
    @napi-rs/triples:  undefined ()
    @next/react-dev-overlay:  undefined ()
    @types/node: 18.0.0 => 18.0.0 
    @types/react: 18.0.14 => 18.0.14 
    @types/react-dom: 18.0.5 => 18.0.5 
    @vercel/nft:  undefined ()
    acorn:  undefined ()
    amphtml-validator:  undefined ()
    arg:  undefined ()
    assert:  undefined ()
    async-retry:  undefined ()
    async-sema:  undefined ()
    aws-amplify: ^4.3.26 => 4.3.26 
    babel-packages:  undefined ()
    browserify-zlib:  undefined ()
    browserslist:  undefined ()
    buffer:  undefined ()
    bytes:  undefined ()
    chalk:  undefined ()
    ci-info:  undefined ()
    cli-select:  undefined ()
    comment-json:  undefined ()
    compression:  undefined ()
    conf:  undefined ()
    constants-browserify:  undefined ()
    content-disposition:  undefined ()
    content-type:  undefined ()
    cookie:  undefined ()
    cross-spawn:  undefined ()
    crypto-browserify:  undefined ()
    cssnano-simple:  undefined ()
    debug:  undefined ()
    devalue:  undefined ()
    domain-browser:  undefined ()
    edge-runtime:  undefined ()
    eslint: 8.18.0 => 8.18.0 
    eslint-config-next: 12.2.0 => 12.2.0 
    etag:  undefined ()
    events:  undefined ()
    find-cache-dir:  undefined ()
    find-up:  undefined ()
    fresh:  undefined ()
    get-orientation:  undefined ()
    glob:  undefined ()
    gzip-size:  undefined ()
    http-proxy:  undefined ()
    https-browserify:  undefined ()
    icss-utils:  undefined ()
    ignore-loader:  undefined ()
    image-size:  undefined ()
    is-animated:  undefined ()
    is-docker:  undefined ()
    is-wsl:  undefined ()
    jest-worker:  undefined ()
    json5:  undefined ()
    jsonwebtoken:  undefined ()
    loader-utils:  undefined ()
    lodash.curry:  undefined ()
    lru-cache:  undefined ()
    micromatch:  undefined ()
    mini-css-extract-plugin:  undefined ()
    nanoid:  undefined ()
    native-url:  undefined ()
    neo-async:  undefined ()
    next: 12.2.0 => 12.2.0 
    node-fetch:  undefined ()
    node-html-parser:  undefined ()
    ora:  undefined ()
    os-browserify:  undefined ()
    p-limit:  undefined ()
    path-browserify:  undefined ()
    postcss-flexbugs-fixes:  undefined ()
    postcss-modules-extract-imports:  undefined ()
    postcss-modules-local-by-default:  undefined ()
    postcss-modules-scope:  undefined ()
    postcss-modules-values:  undefined ()
    postcss-preset-env:  undefined ()
    postcss-safe-parser:  undefined ()
    postcss-scss:  undefined ()
    postcss-value-parser:  undefined ()
    process:  undefined ()
    punycode:  undefined ()
    querystring-es3:  undefined ()
    raw-body:  undefined ()
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    react-is:  17.0.2 
    react-refresh:  0.12.0 
    react-server-dom-webpack:  undefined ()
    regenerator-runtime:  0.13.4 
    sass-loader:  undefined ()
    schema-utils:  undefined ()
    semver:  undefined ()
    send:  undefined ()
    setimmediate:  undefined ()
    source-map:  undefined ()
    stream-browserify:  undefined ()
    stream-http:  undefined ()
    string-hash:  undefined ()
    string_decoder:  undefined ()
    strip-ansi:  undefined ()
    tar:  undefined ()
    terser:  undefined ()
    text-table:  undefined ()
    timers-browserify:  undefined ()
    tty-browserify:  undefined ()
    typescript: 4.7.4 => 4.7.4 
    ua-parser-js:  undefined ()
    unistore:  undefined ()
    util:  undefined ()
    vm-browserify:  undefined ()
    watchpack:  undefined ()
    web-vitals:  undefined ()
    webpack:  undefined ()
    webpack-sources:  undefined ()
    ws:  undefined ()
  npmGlobalPackages:
    @aws-amplify/cli: 7.4.5
    @graphql-codegen/typescript: 2.4.2
    @squoosh/cli: 0.7.2
    aws-cdk-local: 1.65.8
    aws-cdk: 2.29.1
    npm: 6.14.15
```

</details>

### Describe the bug

The Auth modules throws an AuthError when using only identity pool to authorize unauthenticated access. It seems to fallback from checking for federated user to a cognito user - we are not using anything cognito, no user pools - solely identity pools.

### Expected behavior

There should be no errors thrown if authentication is using unauthenticated access with identity pools - and no requirements to have a user pool at all.

### Reproduction steps

1. create an appsync
2. set appsync to use iam access
3. create an identity pool with unauthenticated access
4. provide permissions to appsync from unauth role
5. access appsync from client

### Code Snippet

```javascript
// Put your code below this line.
import { Amplify, API } from "aws-amplify";
import type { NextPage } from "next";
import { useEffect } from "react";

Amplify.configure({
  aws_appsync_graphqlEndpoint:
    "https://APPSYNC_ID.appsync-api.APPSYNC_REGION.amazonaws.com/graphql",
  aws_appsync_region: "us-east-1",
  aws_appsync_authenticationType: "AWS_IAM",
  Auth: {
    identityPoolId: "POOL_ID",
    region: "us-east-1",
  },
});

export const listNotes = `
  query ListNotes {
    listNotes {
      content
    }
  }
`;

const Home: NextPage = () => {
  useEffect(() => {
    const fetchData = async () => {
      const createdTodo = await API.graphql({
        query: listNotes,
        authMode: "AWS_IAM",
      });
      console.log("createdTodo", createdTodo);
    };
    fetchData();
  }, []);

  return <div>notes</div>;
};

export default Home;
```

### Log output

<details>

```
// Put your logs below this line
next-dev.js?3515:24 [ERROR] 10:09.563 AuthError - 
            Error: Amplify has not been configured correctly. 
            The configuration object is missing required auth properties.
            This error is typically caused by one of the following scenarios:

            1. Did you run `amplify push` after adding auth via `amplify add auth`?
                See https://aws-amplify.github.io/docs/js/authentication#amplify-project-setup for more information

            2. This could also be caused by multiple conflicting versions of amplify packages, see (https://docs.amplify.aws/lib/troubleshooting/upgrading/q/platform/js) for help upgrading Amplify packages.
```

</details>

### aws-exports.js

_No response_

### Manual configuration

_No response_

### Additional configuration

_No response_

### Mobile Device

_No response_

### Mobile Operating System

_No response_

### Mobile Browser

_No response_

### Mobile Browser Version

_No response_

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by khatruong2009**
…pool (#10039)

#### Description of changes

Remove error logging on unauthenticated credentials.

#### Issue #, if available

fixes #10039
fixes #9771

#### Description of how you validated changes
Manually tested on sample app

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jamesaucode — APPROVED** [review]
LGTM 🎉

**Comment by jamesaucode** [comment]
> auth unit test › currentUserPoolUser test › get user data error because of user is deleted, disabled or token has been revoked - oAuth case

We should fix this unit test before merging

File: packages/auth/src/Auth.ts [hunk 1]
```diff
@@ -1636,7 +1636,7 @@ export class AuthClass {
 		logger.debug('Getting current session');
 		// Purposely not calling the reject method here because we don't need a console error
```
Comment (helgabalashova) [inline packages/auth/src/Auth.ts:1637]: you should probably remove that comment because now you are actually calling reject method

**Comment by gethinwebster** [comment]
Bumping this: what can we do to get this merged? There are multiple reports of this now: #10035, #10039, #9771, #9845

It looks like it's potentially just waiting for some circleci tasks that have frozen, and might need to be restarted.

**Review by stocaaro — APPROVED** [review]
LGTM 🚢

**Pull request merged**